jsonschema_widget/widgets.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import, unicode_literals
+
+from django import forms
+from django.contrib.admin.templatetags.admin_static import static
+from django.contrib.admin.widgets import AdminTextareaWidget
+from django.urls import reverse
+from django.utils.translation import ugettext_lazy as _
+
+
+class JsonSchemaWidget(AdminTextareaWidget):
+    """
+    JSON Schema Editor widget
+    """
+    @property
+    def media(self):
+        prefix = 'django-netjsonconfig'
+        js = [static('{0}/js/{1}'.format(prefix, f))
+              for f in ('lib/advanced-mode.js',
+                        'lib/tomorrow_night_bright.js',
+                        'lib/jsonschema-ui.js',
+                        'widget.js')]
+        css = {'all': [static('{0}/css/{1}'.format(prefix, f))
+                       for f in ('lib/jsonschema-ui.css',
+                                 'lib/advanced-mode.css')]}
+        return forms.Media(js=js, css=css)
+
+    def render(self, name, value, attrs={}):
+        attrs['class'] = 'vLargeTextField jsoneditor-raw'
+        html = """
+<input class="button json-editor-btn-edit advanced-mode" type="button" value="{0}">
+<script>django._netjsonconfigSchemaUrl = "{1}";</script>
+<label id="netjsonconfig-hint">
+    Want learn to use the advanced mode? Consult the
+    <a href="http://netjsonconfig.openwisp.org/en/stable/general/basics.html"
+       target="_blank">netjsonconfig documentation</a>.
+</label>
+"""
+        html = html.format(_('Advanced mode (raw JSON)'),
+                           reverse('netjsonconfig:schema'))
```

Comment:
5. Since we are moving it out of netjsonconfig, my assumption is that it's planned to be used at other places as well!
Hence, shouldn't all these values be either variables (to be set from the settings file) or changed and made generalised to keep any project in mind!
I am not only talking about the `reverse` here, but also values above are keeping only that one project in mind!

Please correct me if i've understood it incorrectly! :smile:

Reply:
I not sure right now, @nemesisdesign might have better insight than me in this regard.

Reply:
@atb00ker is right, here I explain our main use cases: https://github.com/openwisp/django-jsonschema-widget/issues/2
I think we can work on this in the next iterations, for the moment let's focus on making it work properly on django-netjsonconfig, even if it's dirty (we won't release it yet).